module_utils/network_lsr/nm/client.py

```diff
@@ -4,24 +4,28 @@
 
 # Relative import is not support by ansible 2.8 yet
 # pylint: disable=import-error, no-name-in-module
-from ansible.module_utils.network_lsr.nm.error import LsrNetworkNmError  # noqa:E501
+from ansible.module_utils.network_lsr.nm import error  # noqa:E501
 
 import gi
 
-gi.require_version("NM", "1.0")
+try:
+    gi.require_version("NM", "1.0")
 
-# It is required to state the NM version before importing it
-# But this break the flake8 rule: https://www.flake8rules.com/rules/E402.html
-# Use NOQA: E402 to suppress it.
-from gi.repository import NM  # NOQA: E402
-from gi.repository import GLib  # NOQA: E402
-from gi.repository import Gio  # NOQA: E402
+    # It is required to state the NM version before importing it
+    # But this break the flake8 rule: https://www.flake8rules.com/rules/E402.html
+    # Use NOQA: E402 to suppress it.
+    from gi.repository import NM  # NOQA: E402
+    from gi.repository import GLib  # NOQA: E402
+    from gi.repository import Gio  # NOQA: E402
 
-# pylint: enable=import-error, no-name-in-module
+    # pylint: enable=import-error, no-name-in-module
 
-NM
-GLib
-Gio
+    NM
+    GLib
+    Gio
+    HAS_NM = True
+except (ImportError, ValueError) as e:
+    HAS_NM = False
```

Comment:
This isn't needed to resolve the import problem is it?  I thought this was only needed to set the `HAS_NM` flag, which we don't need, as discussed in https://github.com/linux-system-roles/network/pull/330#issuecomment-747737327

Reply:
This is needed. Without this catch, gi.require_version("NM", "1.0") raises an exception `ValueError: Namespace NM not available` and the module `network_connections` in the task `Configure networking connection profiles` fails. Please see http://pastebin.test.redhat.com/926539.

Reply:
> This is needed. Without this catch, gi.require_version("NM", "1.0") raises an exception `ValueError: Namespace NM not available` and the module `network_connections` in the task `Configure networking connection profiles` fails. Please see http://pastebin.test.redhat.com/926539.

Is it because this module also runs on EL6 where there is no NM?

Reply:
> > This is needed. Without this catch, gi.require_version("NM", "1.0") raises an exception `ValueError: Namespace NM not available` and the module `network_connections` in the task `Configure networking connection profiles` fails. Please see http://pastebin.test.redhat.com/926539.
> 
> Is it because this module also runs on EL6 where there is no NM?

No, this problem occurs when the control node has no NM installed. In our CI test env, the control node fedora 32 has no NM. Thank you for merging this reverting pr [1]. The merge and the deployment undo the NM pre-installation and make this PR testable.

Following the advice from @sivel, "to import the package, not the class" (as described by @rjeffman in https://github.com/linux-system-roles/certificate/pull/71#issuecomment-748051347) fixes the module_utils import failure even with ansible 2.9. So, with this PR 330, we won't see the `__spec__ is None` error any more. But this `gi.require_version("NM", "1.0")` code [0] apparently fails if NM is not installed. Since the code [0] explicitly accesses the package, I think it's reasonable to capture the exception. (Please note that it is safe not to do so on the managed nodes since the package is installed by the network role. This is pure ansible 2.9 collections issue.)

For testing, if we redeploy test-harness having the revert [1] and temporarily remove the try-except catch in this PR, we should be able to see the failure from `gi.require_version("NM", "1.0")` in, e.g., `centos-8/ansible-2.9 (staging)`. Shall we try that?

[0] https://github.com/linux-system-roles/network/pull/330/files#diff-8676715864da9f0f345f37f9a1d253e28e51696897c9fbdd1eede3daf21ba7b5R12

[1] - https://github.com/linux-system-roles/test-harness/pull/136

> I'm trying to understand the conditions (platforms, platform versions) where you get this error.

The condition is the control node, whatever version unless NM is installed...